scripts/transition-staging-repository-state.sh

```diff
@@ -36,48 +34,35 @@ while getopts ":s:ah" opt; do
   esac
 done
 
-# TODO: refactor to use the REST API https://oss.sonatype.org/nexus-staging-plugin/default/docs/index.html to avoid the awkish hell
-# curl --silent -u xxxx:xxx  -H "accept: application/json" -X GET https://s01.oss.sonatype.org/service/local/staging/profile_repositories | jq .
-
-
-# The rc-list table looks like this:
-#[INFO] ID                   State    Description
-#[INFO] iokroxylicious-1032  CLOSED   unknown
-#[INFO] ------------------------------------------------------------------------
-
-STAGING_REPO_IDS=$(mvn ${PLUGIN}:rc-list "${MVN_ARGS[@]}" 2>/dev/null \
-                 | awk '/\[INFO\] ---*/{found=0} found==1{print $2} /\[INFO\] ID.*State.*Description/{found=1}')
+if [[ -z "${STATE}" ]]; then
+    echo "No next state specified. Please specify -s drop|release" 1>&2
+    exit 1
+fi
 
-NUM_REPOS=$(echo "${STAGING_REPO_IDS}" | (grep -c . || true))
+if [[ -z "${DEPLOYMENT_ID}" ]]; then
+    echo "No deployment id specified. Please specify -d <central publishing portal distribution id>" 1>&2
+    exit 1
+fi
 
-if [[ ${ASSERT_NO_STAGING_REPOS} == "true" ]]; then
-  if [[ ${NUM_REPOS} -gt 0 ]]; then
-      >&2 echo "${NUM_REPOS} stage repository(s) already exists but expected none."
-     exit ${NUM_REPOS}
-  fi
-  exit 0
-elif [[ ${STATE} ]]; then
-  if [[ ${NUM_REPOS} -eq 0 ]]; then
-      >&2 echo "No staging repository found to apply a ${STATE} state transition."
-     if [[ ${STATE} == "drop" ]]; then
-         exit 0
-     fi
-     exit 1
-  elif [[ ${NUM_REPOS} -gt 1 ]]; then
-      >&2 echo "Too many staging repositories found (${NUM_REPOS})."
-     exit 1
-  fi
+AUTH=$(printf "${SONATYPE_TOKEN_USERNAME}:${SONATYPE_TOKEN_PASSWORD}" | base64)
```

Comment:
Would it be more secure to do this in the Action and just have `SONTYPE_TOKEN` as the env var? (I'm not clear when secretes are or are not rendered into jobs)